sphinx/builders/__init__.py

```diff
@@ -403,6 +405,109 @@ def build(self, docnames, summary=None, method='update'):
         # wait for all tasks
         self.finish_tasks.join()
 
+    def read(self):
+        # type: () -> List[unicode]
+        """(Re-)read all files new or changed since last update.
+
+        Store all environment docnames in the canonical format (ie using SEP as
+        a separator in place of os.path.sep).
+        """
+        changed, reason = self.env.update_config(self.config, self.srcdir, self.doctreedir)
+
+        # this cache also needs to be updated every time
+        self.env._nitpick_ignore = set(self.config.nitpick_ignore)
+
+        logger.info(bold('updating environment: '), nonl=True)
+
+        self.env.find_files(self.config, self)
+        added, changed, removed = self.env.get_outdated_files(changed)
+
+        # allow user intervention as well
+        for docs in self.app.emit('env-get-outdated', self, added, changed, removed):
```

Comment:
I feel `env-*` events are not good name after moving. But they are used widely by 3rd parties.
So I'm wondering about renaming.